### PR TITLE
Add documentation

### PR DESCRIPTION
**Pull request by RealKC**
* [x] Document public APIs
* [ ] Make docs build fine
* [x] Add a github action to publish my docs to github pages

### CODE REVIEW

**Comment by RealKC** [comment]
Figuring out why stuff that works locally doesn't work on CI is fun....

**Comment by RealKC** [comment]
Ugh, I'll continue this tomorrow

File: docs/symbol.rst [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+================
+
+Reiji's main safety-providing abstraction is the ``symbol<T>`` class.
+
+``symbol<T>`` is a pointer-like proxy that ties a symbol you get from a shared
+library to that library's lifetime. When that library goes out of scope, your
+symbols will become invalid and you will be prevented through runtime mechanisms
+from triggering undefined behaviour.
+
+Throughout this docu ment the term ``invalid`` is used to refer to a symbol on
```
Comment (Kime78) [inline docs/symbol.rst:11]: You accidentally put a space on the word `document`, you wrote `docu ment`

**Review by Kime78 — CHANGES_REQUESTED** [review]
Needs more examples, try addressing these questions: how do I call a method from a symbol, what does reiji::unique_shared_lib contain

**Comment by RealKC** [comment]
Didn't really get anywhere, and I dislike some of `breathe`.

Will eventually revisit with a different approach.

**Pull request closed without merging**